### ISSUE TITLE
Make cython files compatible with Cython3

### ISSUE THREAD

**Issue by kif** [silx-kit/silx#3027]
Cython3, which is currently in alpha phase  does enforce the Python3 syntax.

Because the string management changed so much, there is a big difference and one should be treated manually. I quickly tested and beside a few warnings, there is not much to be done.

**Comment by kif** [comment]
This is declared in the setup.py, not inside the files. my bad.

**Comment by kif** [comment]
Apparently I was not completely drunk when I noticed this ... 
those warning are actually present:
```cythoning silx/io/specfile.pyx to silx/io/specfile.c
/opt/VirtualEnv/py38/lib/python3.8/site-packages/Cython/Compiler/Main.py:344: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /home/jerome/workspace/silx/silx/io/specfile.pyx
```
This does not mean the code is not cython3-compatible but there are a couple of bugs in the build procedure:

* [ ] the cython compiler directives and compile-time options are not taken into account (hence the warning about language_level not set, while it is declared in the setup.py
* [ ] the --force-cython option to `python setup.py build --force-cython` is not working. Nor the environment variable.

**Comment by vallsv** [comment]
Ohhh, bad news. As we did a lot of custom code for the setup, it will be painful. Except if the defaut cython builder provides more features now.

**Comment by t20100** [comment]
BTW, do we still need the `--force-cython`?

**Comment by kif** [comment]
I have a solution ... some part of the cython code is not up to level

**Comment by kif** [comment]
Cython offers now parallel compilation. 

I believe `--force-cython` is a nice feature as the "rm silx/*/*.c" is kind-of dangerous and complicated. If there is a cache, there should always be a way to bypass the cache.

**Comment by t20100** [comment]
CI is broken by cython

**Comment by kif** [comment]
it passed: https://travis-ci.org/github/silx-kit/silx/builds/683740401

**Comment by t20100** [comment]
Sorry, I was talking of CI from other PRs.

Anyway, this is closed by  PR #3034.
